tests/unittest/Core/topomesh.cpp

```diff
@@ -144,5 +292,82 @@ TEST_CASE( "Core/Geometry/TopologicalMesh", "[Core][Core/Geometry][TopologicalMe
         }
         REQUIRE( check1 );
         REQUIRE( check2 );
+        REQUIRE( topologicalMesh.checkIntegrity() );
+    }
+}
+
+void test_split( TopologicalMesh& topo, TopologicalMesh::EdgeHandle eh, float f ) {
+
+    auto he0 = topo.halfedge_handle( eh, 0 );
+    auto he1 = topo.halfedge_handle( eh, 1 );
+    auto v0  = topo.from_vertex_handle( he0 ); // i.e. to_vertex_handle(he1)
+    REQUIRE(v0 == topo.to_vertex_handle(he1));
+    auto v1  = topo.to_vertex_handle( he0 );
+    auto p0  = topo.point( v0 );
+    float f0 = topo.getWedgeData( *( topo.getVertexWedges( v0 ) ).begin() ).m_floatAttrib[0];
+    auto p1  = topo.point( v1 );
+    float f1 = topo.getWedgeData( *( topo.getVertexWedges( v1 ) ).begin() ).m_floatAttrib[0];
+    topo.splitEdgeWedge( eh, f );
+
+    // check validity
+    REQUIRE( topo.is_valid_handle( he0 ) );
+    REQUIRE( topo.is_valid_handle( he1 ) );
+
+    // he0 is untouched
+    REQUIRE( v1 == topo.to_vertex_handle( he0 ) );
+    REQUIRE( Math::areApproxEqual( ( p1 - topo.point( v1 ) ).squaredNorm(), 0_ra ) );
+
+    // he1 point to inserted vertex
+    auto vsplit = topo.to_vertex_handle( he1 ); // i.e. from_vertex_handle(he0)
+    REQUIRE(vsplit == topo.from_vertex_handle(he0));
+
+    auto psplit = topo.point( vsplit );
+    auto vcheck = ( f * p1 + ( 1.f - f ) * p0 );
+    REQUIRE( Math::areApproxEqual( ( psplit - vcheck ).squaredNorm(), 0.f ) );
+
+    auto wedges = topo.getVertexWedges( vsplit );
+    REQUIRE( wedges.size() == 1 );
+
+    auto wd     = topo.getWedgeData( *wedges.begin() );
+    auto fsplit = wd.m_floatAttrib[0];
+    auto fcheck = ( f * f1 + ( 1.f - f ) * f0 );
+    REQUIRE( Math::areApproxEqual( fsplit, fcheck ) );
+    REQUIRE( Math::areApproxEqual( ( psplit - wd.m_position ).squaredNorm(), 0.f ) );
+}
+
+TEST_CASE( "Core/Geometry/TopologicalMesh/EdgeSplit", "[Core][Core/Geometry][TopologicalMesh]" ) {
+    using Ra::Core::Vector3;
+    using Ra::Core::Geometry::TopologicalMesh;
+    using Ra::Core::Geometry::TriangleMesh;
+
+    // create a triangle mesh with 4 vertices
+    TriangleMesh meshSplit;
+    meshSplit.setVertices( {{0, 0, 0}, {1, 0, 0}, {1, 1, 0}, {0, 1, 0}} );
+    meshSplit.setNormals( {{-1, -1, 1}, {1, -1, 1}, {1, 1, 1}, {-1, 1, 1}} );
+    meshSplit.m_indices = {Vector3ui( 0, 1, 2 ), Vector3ui( 0, 2, 3 )};
+    // add a float attrib
+    auto handle = meshSplit.addAttrib<float>( "test", {0.f, 1.f, 2.f, 3.f} );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'handle' is assigned a value that is never used.](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=4930613)